preprocessing/nextclade/src/loculus_preprocessing/embl.py

```diff
@@ -0,0 +1,261 @@
+import io
+import logging
+from typing import Any
+
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqFeature import CompoundLocation, FeatureLocation, Reference, SeqFeature
+from Bio.SeqRecord import SeqRecord
+
+from .config import Config
+
+logger = logging.getLogger(__name__)
+
+# EMBL allowed qualifiers constant
+EMBL_ANNOTATIONS: dict[str, list[str]] = {
+    "cds_qualifiers": [
+        "allele",
+        "artificial_location",
+        "circular_RNA",
+        "codon_start",
+        # "db_xref",  # protein accession of reference
+        "EC_number",
+        "exception",
+        "experiment",
+        "function",
+        "gene",
+        "gene_synonym",
+        "inference",
+        # "locus_tag",  # must be pre-registered with ENA
+        # "old_locus_tag",
+        "map",
+        "note",
+        "number",
+        "operon",
+        "product",
+        "protein_id",
+        "pseudo",
+        "pseudogene",
+        "ribosomal_slippage",
+        "standard_name",
+        "translation",
+    ],
+    "gene_qualifiers": [
+        "allele",
+        # "db_xref",
+        "experiment",
+        "function",
+        "gene",
+        "gene_synonym",
+        "inference",
+        # "locus_tag",
+        # "old_locus_tag",
+        "map",
+        "note",
+        "operon",
+        "product",
+        "pseudo",
+        "pseudogene",
+        "pseudotype",
+        "standard_name",
+        "trans_splicing",
+    ],
+}
+
+
+def get_country(metadata: dict[str, str], config: Config) -> str:
+    country = metadata.get(config.embl_info.country_property, "Unknown")
+    admin_levels = config.embl_info.admin_level_properties
+    admin = ", ".join([metadata.get(level) for level in admin_levels if metadata.get(level)])  # type: ignore
+    return f"{country}: {admin}" if admin else country
+
+
+def get_description(accession, version, db_name) -> str:
+    return (
+        f"Original sequence submitted to {db_name} with accession: "
+        f"{accession}, version: {version}"
+    )
+
+
+def reformat_authors_from_loculus_to_embl_style(authors: str) -> str:
+    authors_list = [author for author in authors.split(";") if author]
+    ena_authors = []
+    for author in authors_list:
+        last_names, first_names = author.split(",")[0].strip(), author.split(",")[1].strip()
+        initials = "".join([name[0] + "." for name in first_names.split() if name])
+        ena_authors.append(f"{last_names} {initials}".strip())
+    return ", ".join(ena_authors) + ";"
```

Comment:
The ";" is now duplicated in the embl-flat files because Bio is adding a ";" as well - as the authors are only used for the embl flat file here we should just remove the ";"
```suggestion
    return ", ".join(ena_authors)
```